### PR TITLE
Update build script and include more platforms

### PR DESCRIPTION
**Pull request by andreasnoack**

### CODE REVIEW

File: build_tarballs.jl [hunk 1]
```diff
@@ -1,52 +1,50 @@
...
-if !isempty(get(ENV,"TRAVIS_TAG",""))
-    print_buildjl(pwd(), products, hashes,
-        "https://github.com/KristofferC/OpenspecfunBuilder/releases/download/$(ENV["TRAVIS_TAG"])")
-end
+# Dependencies that must be installed before this package can be built
+dependencies = [
+]
 
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, "libRmath", sources, script, platforms, products, dependencies)
```
Comment (stevengj) [inline build_tarballs.jl]: libRmath?

**Comment by stevengj** [comment]
I incorporated most of these changes into #7.

**Comment by andreasnoack** [comment]
Why not just update this PR? Is your addition to the script to make it run only when a release is tagged?

I realized that FreeBSD is supported on the latest version of BinaryBuilder so I've enabled all platforms. However, it looks like something has changed in Windows. @staticfloat Has the Windows compiler changed?

**Comment by staticfloat** [comment]
> Has the Windows compiler changed?

No, but [the latest CI run that I found](https://travis-ci.org/KristofferC/OpenspecfunBuilder) is failing on windows because of [this line](https://github.com/KristofferC/OpenspecfunBuilder/commit/10a89dddcf0e3799171a84f2aa1d6592b5551a6b#diff-2567705d891d25120e6ea4ca97383113R19), which is failing on Windows due to the fact that the Openspecfun build script isn't recognizing that it should be naming things as `.dll` instead of `.so`.  One of the things that _has_ changed in BinaryBuilder is that bash is run with `set -e`, meaning it will stop if a command fails.  So if you want to ignore errors, you have to explicitly ignore them (e.g. `|| true` at the end).

In this case, I think it's best to just force the Openspecfun buildsystem to realize it's on windows.  Probably by doing something like `make OS=WINNT` or something like that.

**Comment by andreasnoack** [comment]
Thanks for the explanation. I like that the whole run fails if a single platforms fails. It makes it easier to detect errors.

I think I finally managed to get this one working for all supported platforms.

**Comment by stevengj** [comment]
Sorry, @andreasnoack, I didn't notice your PR at first when I created mine.  I can merge some changes in the other direction if you want, though I don't have permissions on this branch I think?

The addition to the Travis script is so that, on a tag, it re-runs with --only-buildjl afterwards.  This is only needed if we end up using the `--part` flag to split the build into parts, though; I haven't turned that on yet since I wasn't sure how long the build would take.

**Comment by KristofferC** [comment]
This was mostly an experiment from my side in using BinaryBuilder but if it works well now with the updates here, perhaps we should move the repo to some org?

**Comment by andreasnoack** [comment]
I guess JuliaMath is the right place. It's where `SpecialFunctions.jl` lives.

**Comment by KristofferC** [comment]
Alright, "You don’t have the permission to create repositories on JuliaMath" though.

**Comment by andreasnoack** [comment]
Great. It went through. I'll merge this one and try to update the deployment info to match the new home. @stevengj Could you rebase your PR with this to get the last changes?

**Pull request merged**